### ISSUE TITLE
TypelevelAs suggests impossible changes

### ISSUE THREAD

**Issue by NeQuissimus** [typelevel/typelevel-scalafix#49]
Version 0.1.5

```scala
import cats.effect.Resource

val foo: Resource[A, B] = ???
val x = foo.map(_ => ())
```

The rule `TypelevelAs.as` will complain and suggest to rewrite the `map` line to `.void`.
`Resource` does not have a `void` method :)
The check should probably be limited to types that provide the necessary functions.

**Comment by armanbilge** [comment]
It does, if you do `import cats.syntax.all._` :) I think there's a way to scalafix to add imports if necessary.

**Comment by NeQuissimus** [comment]
Oh, that is possible. I wonder if there is a good way to make that part of the message in such a situation maybe?

My actual issue turned out to be that I had multiple `Functor` instances (from `Temporal` and `Sync`) and the void syntax could not decide which one to pick :)

**Comment by armanbilge** [comment]
Here's how an http4s scalafix adds an import if needed.
https://github.com/typelevel/typelevel-scalafix/blob/8e246387dbed1e772933d2f757bc7609bc909861/modules/http4s/rules/src/main/scala/org/typelevel/fix/Http4sLiteralsSyntax.scala#L41-L43

**Comment by NeQuissimus** [comment]
Let's close this issue because there is no issue :)

**Comment by armanbilge** [comment]
Actually there is an issue, the import is not being added :)

**Comment by NeQuissimus** [comment]
Right but you'd have to make the rule a lot more specific then, would you not?
What would happen to a `foo.map(_ => ())` where `foo` does not have a `void` in `cats.syntax`?

**Comment by armanbilge** [comment]
Good point. So even after adding the import, this Scalafix could still be suggesting broken code. Seems to me like another reason to keep it open.

**Comment by NeQuissimus** [comment]
👍 I agree with that. This almost needs something like "do we have an implicit Functor? OK, just add the syntax" and otherwise bail out.
